Dockerfile

```diff
@@ -1,14 +1,121 @@
+RUN set -xe \
+	&& OTP_DOWNLOAD_URL="https://github.com/erlang/otp/releases/download/OTP-${BUILD_VERSION}/otp_src_${BUILD_VERSION}.tar.gz" \
+	&& OTP_DOWNLOAD_SHA256="${BUILD_SHA256}" \
+	&& fetchDeps=' \
+		curl \
+		ca-certificates' \
+	&& apt-get update \
+	&& apt-get install -y --no-install-recommends $fetchDeps \
+	&& curl -fSL -o otp-src.tar.gz "$OTP_DOWNLOAD_URL" \
+	&& echo "$OTP_DOWNLOAD_SHA256  otp-src.tar.gz" | sha256sum -c - \
+	&& runtimeDeps=' \
+		libssl1.1 \
+	' \
+	&& buildDeps=' \
+		autoconf \
+		dpkg-dev \
+		gcc \
+		g++ \
+		make \
+		libncurses-dev \
+		libssl-dev \
+	' \
+	&& apt-get install -y --no-install-recommends $runtimeDeps \
+	&& apt-get install -y --no-install-recommends $buildDeps \
+	&& export ERL_TOP="/usr/src/otp_src_${OTP_VERSION%%@*}" \
+	&& mkdir -vp $ERL_TOP \
+	&& tar -xzf otp-src.tar.gz -C $ERL_TOP --strip-components=1 \
+	&& rm otp-src.tar.gz \
+	&& ( cd $ERL_TOP \
+	  && ./otp_build autoconf \
+	  && gnuArch="$(dpkg-architecture --query DEB_HOST_GNU_TYPE)" \
+	  && ./configure --build="$gnuArch" \
+	  && make -j$(nproc) \
+	  && make install ) \
+	&& find /usr/local -name examples | xargs rm -rf \
+	&& REBAR3_DOWNLOAD_URL="https://github.com/erlang/rebar3/archive/${REBAR3_VERSION}.tar.gz" \
+	&& REBAR3_DOWNLOAD_SHA256="a14711b09f6e1fc1b080b79d78c304afebcbb7fafed9d0972eb739f0ed89121b" \
+	&& mkdir -p /usr/src/rebar3-src \
+	&& curl -fSL -o rebar3-src.tar.gz "$REBAR3_DOWNLOAD_URL" \
+	&& echo "$REBAR3_DOWNLOAD_SHA256 rebar3-src.tar.gz" | sha256sum -c - \
+	&& tar -xzf rebar3-src.tar.gz -C /usr/src/rebar3-src --strip-components=1 \
+	&& rm rebar3-src.tar.gz \
+	&& cd /usr/src/rebar3-src \
+	&& HOME=$PWD ./bootstrap \
+	&& install -v ./rebar3 /usr/local/bin/ \
+	&& rm -rf /usr/src/rebar3-src \
+	&& apt-get purge -y --auto-remove $buildDeps $fetchDeps \
+	&& rm -rf $ERL_TOP /var/lib/apt/lists/*
+
+ARG BUILD_TARGET=docker_rosetta
+
+RUN set -xe \
+	    && apt update \
+	    && apt-get install -y --no-install-recommends \
+     	        libssl-dev make automake autoconf libncurses5-dev gcc \
+	        libdbus-1-dev libbz2-dev bison flex libgmp-dev liblz4-dev \
+	        libsodium-dev sed wget curl build-essential libtool git \
+		ca-certificates \
+	    && mkdir -p /opt/cmake \
+	    && wget -O /opt/cmake/cmake.tgz \
+	        https://github.com/Kitware/CMake/releases/download/v3.21.3/cmake-3.21.3-linux-x86_64.tar.gz \
+            && tar -zxf /opt/cmake/cmake.tgz -C /opt/cmake --strip-components=1
+
+# Install Rust toolchain
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+ENV CC=gcc CXX=g++ CFLAGS="-U__sun__" \
+    ERLANG_ROCKSDB_OPTS="-DWITH_BUNDLE_SNAPPY=ON -DWITH_BUNDLE_LZ4=ON" \
+    ERL_COMPILER_OPTIONS="[deterministic]" \
+    PATH="/root/.cargo/bin:/opt/cmake/bin:$PATH" \
+    RUSTFLAGS="-C target-feature=-crt-static"
+
+WORKDIR /usr/src/
 
+# Add our code
+RUN git clone https://github.com/helium/blockchain-node.git
```

Comment:
This seems like overkill. if a user wants to do a full build, they can pass in a separate container as a base container right?

Reply:
Hm I'm not entirely tracking here. From my understanding we need the Dockerfile node to build from source without any dependencies. Wouldn't passing in other containers be considered an outside dependency?

Reply:
I think @syuan100 is correct, we have to build from source as we can't trust any base container that includes the dependencies.

Reply:
@GeekArthur fair but if you're providing the build image from another source as a build arg, that satisfies your requirements too right? 

My proposal is that you run two builds:

1. blockchain-node (like we do and publish to your local docker repo)
2. rosetta-helium (and you pass in your reference as a UBUNTU_RUNNER). 

That keeps this repo's Dockerfile smaller and doesn't require a full rebuild if folks want to "trust" the helium core team's UBUNTU_RUNNER image.

Reply:
I see the value in this approach. However I do also see that in the Rosetta docs it is required to be able to run all components from just one Dockerfile. Maybe we can add a `Dockerfile_quick` to allow for bypassing a from scratch build and utilize the other image builds for speed.

Reply:
I think I will also try to rework the main Dockerfile to take an arg for testnet as opposed to having a separate Dockerfile.

Reply:
Okay. I'm convinced by the rosetta docs argument!

Reply:
@abhay Yeah, we have to follow the security rules from Rosetta, but thanks for the proposal.